integration/ctr_metrics_test.go

```diff
@@ -0,0 +1,235 @@
+
+	params[paramFrequency] = "2s"
+	invalidAction := "invalidRequest"
+	_, err = util.ExecuteOperation(suite.Cfg, suite.metricsUrl, invalidAction, params)
+	assert.Errorf(suite.T(), err, "error while sending metrics request with wrong topic %v", invalidAction)
+}
+
+func (suite *ctrMetricsSuite) assertContainerMetricsFeatures() {
+	_, err := util.SendDigitalTwinRequest(suite.Cfg, http.MethodGet, suite.metricsUrl, nil)
+	require.NoError(suite.T(), err, "error while getting the metrics feature")
+}
+
+func (suite *ctrMetricsSuite) stopMetricsRequest() {
+	suite.executeMetrics(stopParams)
+}
+
+func (suite *ctrMetricsSuite) executeMetrics(params map[string]interface{}) {
+	_, err := util.ExecuteOperation(suite.Cfg, suite.metricsUrl, actionRequest, params)
+	assert.NoErrorf(suite.T(), err, "error while sending metrics request for containers with params %v", params)
+}
+
+func (suite *ctrMetricsSuite) testMetrics(params map[string]interface{}, expectedOriginators ...string) error {
+	wsConnection, err := util.NewDigitalTwinWSConnection(suite.Cfg)
+	defer wsConnection.Close()
+	require.NoError(suite.T(), err, "failed to create websocket connection")
+
+	err = util.SubscribeForWSMessages(suite.Cfg, wsConnection, "START-SEND-MESSAGES", "")
+	defer suite.stopMetricsRequest()
+	require.NoError(suite.T(), err, "unable to listen for events by using a websocket connection")
+
+	timestamp := time.Now().Unix()
+	actualOriginators := make(map[string]bool)
+
+	suite.executeMetrics(params)
+
+	result := util.ProcessWSMessages(suite.Cfg, wsConnection, func(msg *protocol.Envelope) (bool, error) {
+		if msg.Path != suite.pathData {
+			return false, nil
+		}
+
+		if msg.Topic.String() != suite.topicData {
+			return false, nil
+		}
+
+		data, err := json.Marshal(msg.Value)
+		if err != nil {
+			return true, err
+		}
+
+		metric := new(things.MetricData)
+		if err := json.Unmarshal(data, metric); err != nil {
+			return true, err
+		}
+
+		if metric.Timestamp < timestamp {
+			return true, fmt.Errorf("Invalid timestamp: %v", metric.Timestamp)
+		}
+
+		for _, m := range metric.Snapshot {
+			for _, originator := range expectedOriginators {
+				if originator == m.Originator {
+					actualOriginators[originator] = true
+					break
+				}
+			}
+
+			if _, ok := actualOriginators[m.Originator]; !ok {
+				return true, fmt.Errorf("Invalid originator: %s", m.Originator)
+			}
+
+			for _, mm := range m.Measurements {
+				if strings.HasPrefix(mm.ID, "cpu.") {
+					continue
+				}
+
+				if strings.HasPrefix(mm.ID, "memory.") {
+					continue
+				}
+
+				if strings.HasPrefix(mm.ID, "io.") {
+					continue
+				}
+
+				if strings.HasPrefix(mm.ID, "net.") {
+					continue
+				}
+
+				if strings.HasPrefix(mm.ID, "pids") {
+					continue
+				}
+
+				return true, fmt.Errorf("Invalid metrics ID: %s", mm.ID)
+			}
+		}
+
+		return len(expectedOriginators) == len(actualOriginators), nil
+	})
+
+	return result
+}
```

Comment:
please refactor it something like this:
```suggestion
				if !allowedPrefixID(mm.ID) {
					return true, fmt.Errorf("Invalid metrics ID: %s", mm.ID)
				}
			}
		}

		return len(expectedOriginators) == len(actualOriginators), nil
	})

	return result
}

func allowedPrefixID(id string) bool {
	allowedIDs := []string{"cpu.", "memory.", "io.", "net.", "pids"}
	for _, allowedID := range allowedIDs {
		if strings.HasPrefix(id, allowedID){
			return true
		}
	}
	return false
}
```